kafka/conn.py

```diff
@@ -78,6 +78,13 @@ class SSLWantWriteError(Exception):
     gssapi = None
     GSSError = None
 
+# needed for setblocking()
+try:
+    from gevent import monkey
+    is_gevent_patched = monkey.is_module_patched('ssl')
```

Comment:
Can you rename `is_gevent_patched` to clarify that you're only checking whether `ssl` is patched? Maybe `ssl_is_patched_by_gevent`...

Otherwise it's ambiguous whether you're checking for any monkey-patching at all...  Also, FYI monkey-patching threading/sockets can lead to surprising behavior on the background heartbeat thread.

Reply:
I renamed the flag as requested.
Regarding heartbeat, I haven't looked at the implementation, but from my experience with gevent monkey-patching, it works quite well with threading and sockets.
Only rarely have I encountered problems with these modules.

Reply:
Here's more details on the potential problem with gevent and the background heartbeat: https://github.com/dpkp/kafka-python/issues/1515#issuecomment-397147496

Reply:
Well, it does make sense that in a gevent monkey-patched env, if the main thread loop is CPU bound (haven't looked at the code so I just take your word for it) it will not yield to other coroutines, unless yielding is done explicitly.
I think that given the popularity of gevent and its monkey-patching capabilities, it would be fair to handle this situation in the main loop.
Maybe do a check to see if the threading module is patched, and if so, `gevent.sleep(0)` during the main loop.
If this is ok with you, let me know and I'll add that to this PR.